flink-core/src/main/java/org/apache/flink/util/FileUtils.java

```diff
@@ -525,6 +540,123 @@ public static Path expandDirectory(Path file, Path targetDirectory) throws IOExc
 		return new Path(targetDirectory, rootDir);
 	}
 
+	/**
+	 * List the {@code directory} recursively and return the files that satisfy the {@code fileFilter}.
+	 *
+	 * @param directory the directory to be listed
+	 * @param fileFilter a file filter
+	 * @return a collection of {@code File}s
+	 *
+	 * @throws IOException if an I/O error occurs while listing the files in the given directory
+	 */
+	public static Collection<File> listFilesInPath(final File directory, final Predicate<File> fileFilter) throws IOException {
+		checkNotNull(directory, "directory");
+		checkNotNull(fileFilter, "fileFilter");
+
+		if (!Files.exists(directory.toPath())) {
+			throw new IllegalArgumentException(String.format("The directory %s dose not exist.", directory));
+		}
+		if (!Files.isDirectory(directory.toPath())) {
+			throw new IllegalArgumentException(String.format("The %s is not a directory.", directory));
+		}
+
+		final FilterFileVisitor filterFileVisitor = new FilterFileVisitor(fileFilter);
+
+		Files.walkFileTree(
+			directory.toPath(),
+			EnumSet.of(FileVisitOption.FOLLOW_LINKS),
+			Integer.MAX_VALUE,
+			filterFileVisitor);
+
+		return filterFileVisitor.getFiles();
+	}
+
+	/**
+	 * Convert a collection of {@code File}s to a collection of relative path to the working dir.
+	 *
+	 * @param files a collection of files needed to be relatived
+	 * @return a collection of relative {@code File}s
+	 */
+	public static Collection<File> relativizeToWorkingDir(final Collection<File> files) {
+		checkNotNull(files, "files");
+
+		if (files.isEmpty()) {
+			return Collections.emptyList();
+		}
+
+		final java.nio.file.Path workingDirPath = Paths.get(System.getProperty("user.dir"));
+
+		final List<File> relativeFiles = new LinkedList<>();
+
+		for (File file : files) {
+			if (file.isAbsolute()) {
+				relativeFiles.add(workingDirPath.relativize(file.toPath()).toFile());
+			} else {
+				relativeFiles.add(file);
+			}
+		}
+
+		return Collections.unmodifiableCollection(relativeFiles);
+	}
+
+	/**
+	 * Convert a collection of relative {@code File}s to a collection of relative {@code URL}s.
+	 *
+	 * @param relativeFiles a collection of relative {@code File}s
+	 * @return a collection of relative URLs
+	 *
+	 * @throws MalformedURLException if error occurs while construct a url.
+	 */
+	public static Collection<URL> toRelativeURLs(final Collection<File> relativeFiles) throws MalformedURLException {
```

Comment:
Wouldn't it be enough to have a method `URL toURL(File)` which transforms a `File` into a `URL`? The `File` will then decide whether the URL is relative or absolute, depending on the `File`.